internal/parse/parser.go

```diff
@@ -187,57 +215,105 @@ func isNameByte(c byte) bool {
+	if p.skipName() {
+		return p.input[idStart:p.pos], true
+	}
+	return "", false
+}
 
-		{bytef: p.skipByteFind, result: []bool{false}, input: "", data: []string{"a"}},
-		{bytef: p.skipByteFind, result: []bool{false, true, true}, input: "abcde", data: []string{"x", "b", "c"}},
-		{bytef: p.skipByteFind, result: []bool{true, false}, input: "abcde ", data: []string{" ", " "}},
+// parseFullName parses a column name or Go field name, optionally dot-prefixed by
+// its table or type name respectively.
+func (p *Parser) parseFullName(idc idClass) (FullName, bool) {
+	cp := p.save()
+	var fn FullName
+	p.skipSpaces()
+	if id, ok := p.parseIdentifier(); ok {
+		fn.Prefix = id
+		if p.skipByte('.') {
+			if id, ok := p.parseIdentifier(); ok {
+				fn.Name = id
+				return fn, true
+			}
+		} else {
+			// A column name specified without a table prefix is a name not a
+			// prefix
+			if idc == columnId {
+				fn.Name = fn.Prefix
+				fn.Prefix = ""
+			}
+			return fn, true
+		}
+	}
+	cp.restore()
+	return fn, false
+}
 
-		{stringf0: p.skipSpaces, result: []bool{false}, input: "", data: []string{}},
-		{stringf0: p.skipSpaces, result: []bool{false}, input: "abc    d", data: []string{}},
-		{stringf0: p.skipSpaces, result: []bool{true}, input: "     abcd", data: []string{}},
-		{stringf0: p.skipSpaces, result: []bool{true}, input: "  \t  abcd", data: []string{}},
-		{stringf0: p.skipSpaces, result: []bool{false}, input: "\t  abcd", data: []string{}},
+// parseInputExpression parses an SDL input go-defined type to be used as a
+// query argument.
+func (p *Parser) parseInputExpression() (*InputPart, bool, error) {
+	cp := p.save()
+	var err error
+	var fn FullName
+	var ok bool
 
-		{stringf: p.skipString, result: []bool{false}, input: "", data: []string{"a"}},
-		{stringf: p.skipString, result: []bool{true, true}, input: "helloworld", data: []string{"hElLo", "w"}},
-		{stringf: p.skipString, result: []bool{true, true}, input: "hello world", data: []string{"hello", " "}},
+	p.skipSpaces()
+	if p.skipByte('$') {
+		fn, ok = p.parseFullName(typeId)
+		if !ok {
+			err = fmt.Errorf("malformed input type")
+		}
+		p.skipSpaces()
+		return &InputPart{fn}, true, err
 	}
-	for _, v := range parseTests {
-		p.Parse(v.input)
-		for i, _ := range v.result {
-			var result bool
-			if v.bytef != nil {
-				result = v.bytef(v.data[i][0])
-			}
-			if v.stringf != nil {
-				result = v.stringf(v.data[i])
-			}
-			if v.stringf0 != nil {
-				result = v.stringf0()
-			}
-			if v.result[i] != result {
-				log.Printf("Test %#v failed. Expected: '%t', got '%t'\n", v, result, v.result[i])
+	cp.restore()
+	return nil, false, nil
+}
+
+// parseInputExpression parses an SDL input go-defined type to be used as a
+// query argument.
+func (p *Parser) parseStringLiteral() (*BypassPart, bool, error) {
+	cp := p.save()
+
+	var err error
+
+	if p.pos < len(p.input) {
+		c := p.input[p.pos]
+		if c == '"' || c == '\'' {
+			p.skipByte(c)
+			if !p.skipByteFind(c) {
```

Comment:
It's okay to ignore this for now, but we need at the very least a `// TODO Handle escaping` here.

Reply:
I'll add it to our Jira board too